lib/nimble_options/docs.ex

```diff
@@ -257,6 +257,9 @@ defmodule NimbleOptions.Docs do
 
       {:struct, _struct_name} ->
         quote(do: atom())
+
+      {:tuple, subtypes} ->
+        subtypes |> Enum.map(&type_to_spec/1) |> List.to_tuple()
```

Comment:
This is not going to work for tuples other than 2-element tuples, since the AST for those is different.

```elixir
iex(2)> quote do: {1, 2, 3}
{:{}, [], [1, 2, 3]}
iex(3)> quote do: {1, 2}
{1, 2}
iex(4)> quote do: {1}
{:{}, [], [1]}
```

I think something like this should work:

```suggestion
        subtypes |> Enum.map(&type_to_spec/1) |> List.to_tuple() |> Macro.escape()
```

Let's add a test for a type like `{:tuple, [{:list, :integer}]}` or something like this to strengthen the tests.